File: in_typescript/src/Pattern_19.ts
export default function pattern_19() {
  let row = 5;
  let column = 9;
  let symbol = "A";
  const halfColumn = Math.floor(column / 2);

  for (let i = 0; i < row; i++) {
    symbol = "A";
    for (let j = 0; j < column; j++) {
      if (j >= halfColumn - i && j <= halfColumn + i) {
        process.stdout.write(`${symbol}`);
        let s = symbol.charCodeAt(0);
        symbol = String.fromCharCode(j < halfColumn ? s + 1 : s - 1);
      } else {
        process.stdout.write(` `);
      }
    }
    process.stdout.write(`\n`);
  }
}

File: in_rust/src/pattern_19.rs
pub fn pattern_19() {
    let row = 5;
    let column = 9;
    let mut symbol;
    let half_column = column / 2;

    for i in 0..row {
        symbol = 'A';
        for j in 0..column {
            if j >= half_column - i && j <= half_column + i {
                print!("{}", symbol);
                if j < half_column {
                    let s = symbol as u8 + 1;
                    symbol = s as char;
                } else {
                    let s = symbol as u8 - 1;
                    symbol = s as char;
                }
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
